File: Java/src/com/leetcode/Medium/BitwiseANDOfNumbersRange.java
package com.leetcode.Medium;

// 201. Bitwise AND of Numbers Range
public class BitwiseANDOfNumbersRange {
    public int rangeBitwiseAnd(int m, int n) {
        int i = 0;
        while ((i | m) < n) {
            i = (i << 1) | 1;
        }

        return m & ~i;
    }
}

File: JavaScript/Medium/BitwiseAndOfNumbersRange.js
/**
 * 201. Bitwise AND of Numbers Range
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
const rangeBitwiseAnd = function (left, right) {
  let i = 0;
  while ((i | left) < right) {
    i = (i << 1) | 1;
  }

  return left & ~i;
};